### ISSUE TITLE
better bit-wrangling abstraction

### ISSUE THREAD

**Issue by Krastanov** [QuantumSavory/QuantumClifford.jl#113]
In many places in the code you can see

```
    T = eltype(xzs)
    lowbit = T(1)
    ibig = _div(T,i-1)+1
    ismall = _mod(T,i-1)
    ismallm = lowbit<<(ismall)
```

where `ibig` is the index within the array of UIntx and `ismall` is the bit within that UIntx. This is getting too repetitive at this point. Find a better way to write it, abstract it away.

Consult the small symbolic Cliffords (e.g. `sHadamard`) as they have some more abstraction.

**Comment by Krastanov** [comment]
@amicciche  , this is related to your pull request

**Comment by Krastanov** [comment]
Keeping this open as there can still be additional cleanup even after #367.

#367 would be a good reference for future work on this as well.

In particular, now with #364 we consistently just get the small and big indices, but an abstraction that does not require the user to explicitly perform that step would still be useful. E.g. something like `bi = BitIndexer(stabilizer); bi[i,j] == stabilizer.xzs[ibig] & imask` (up to some typos in the example). But really, it would be much better if just writing `stabilizer[i,j]` ends up being as fast so we do not need this explicit indexing.
